crates/btcio/src/reader/ops_visitor.rs

```diff
@@ -0,0 +1,316 @@
+use digest::Digest;
+use sha2::Sha256;
+use strata_l1tx::filter::visitor::OpsVisitor;
+use strata_state::{
+    batch::SignedBatchCheckpoint,
+    tx::{DepositInfo, DepositRequestInfo, ProtocolOperation},
+};
+
+/// Ops visitor for rollup client. This is intended to get Da commitment as well as process/store da
+/// blob.
+#[derive(Clone, Debug)]
+pub struct ClientOpsVisitor {
+    ops: Vec<ProtocolOperation>,
+    // TODO: Add l1 manager to store da to db
```

Comment:
This would be increasing coupling.  This should be done as a separate stage *after* we've run the initial pass.